src/Libraries/Microsoft.Extensions.AI.Evaluation.NLP/BLEU/RationalNumber.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+
+namespace Microsoft.Extensions.AI.Evaluation.NLP.BLEU;
+
+internal readonly struct RationalNumber : IEquatable<RationalNumber>
+{
+    public RationalNumber(int numerator, int denominator)
+    {
+        if (denominator == 0)
+        {
+            throw new DivideByZeroException("Denominator cannot be zero.");
+        }
+
+        Numerator = numerator;
+        Denominator = denominator;
+    }
+
+    public int Numerator { get; }
+    public int Denominator { get; }
+
+    public double ToDouble() => (double)Numerator / Denominator;
```

Comment:
This seems a little odd. Should this be a `Value` property instead? Also, should we cache the result since the it is not expected to change?

Reply:
I kind of prefer the `ToDouble()` since it makes the type conversion semantics clear. I don't see any need to cache the value as it is only accessed typically once, and sometimes it won't be computed at all.